### PR TITLE
Swap CtyInstanceState with PlanResourceChange flag

### PR DESCRIPTION
**Pull request by t0yv0**
An experiment to see if PlanResourceChange can substitute for CtyInstanceState while still handling wafv2_web_acl resource correctly. See https://github.com/pulumi/pulumi-aws/issues/3361

### CODE REVIEW

**Comment by t0yv0** [comment]
This slightly regressed on the ability to cleanly refresh the state for the example program  https://github.com/pulumi/pulumi-aws/issues/3361  - but the unclean refresh now matches TF actual changes. It's in a broad category of issues of this sort, we need to figure out the policy for it. So I slightly am leaning on recommending to merge this in so we can retire CtyInstanceState.

**Comment by VenelinMartinov** [comment]
The unclean refresh is rather unfortunate but probably not a big deal.
I think I saw somewhere that it only happens once and then doesn't happen during subsequent up/refreshes, is this correct?

Does refresh run Diff? Could we add some diff customization for these properties to say null and empty are the same?

**Review by VenelinMartinov — APPROVED** [review]

File: provider/resources.go [hunk 1]
```diff
@@ -733,7 +733,14 @@ func ProviderFromMeta(metaInfo *tfbridge.MetadataInfo) *tfbridge.ProviderInfo {
 	v2p := shimv2.NewProvider(upstreamProvider.SDKV2Provider,
 		shimv2.WithDiffStrategy(shimv2.PlanState),
 		shimv2.WithPlanResourceChange(func(s string) bool {
-			return s == "aws_ssm_document"
+			switch s {
+			case "aws_ssm_document":
+				return true
+			case "aws_wafv2_web_acl":
```
Comment (iwahbe) [inline provider/resources.go:739]: Style nit:
```suggestion
			case "aws_ssm_document", "aws_wafv2_web_acl":
```

**Review by iwahbe — APPROVED** [review]
Let's retire `CtyInstanceState`.

**Comment by t0yv0** [comment]
> I think I saw somewhere that it only happens once and then doesn't happen during subsequent up/refreshes, is this correct?

Yes.

I still think we might want to work that one out to be more friendlier but it is indeed somewhat tolerable.

**Pull request merged**